mm/ktsan/stack_depot.c

```diff
@@ -0,0 +1,132 @@
+#include "ktsan.h"
+
+#include <linux/jhash.h>
+#include <linux/kernel.h>
+
+#define KT_STACK_HASH_SEED 0x9747b28c
+
+/* Only available during early boot. */
+void __init kt_stack_depot_init(kt_stack_depot_t *depot,
+			unsigned part_count, unsigned stack_count)
+{
+	unsigned i;
+
+	depot->part_count = part_count;
+	depot->stack_count = stack_count;
+
+	kt_cache_init(&depot->part_cache,
+			sizeof(*depot->parts) * part_count, 1);
+	depot->parts = kt_cache_alloc(&depot->part_cache);
+	BUG_ON(depot->parts == NULL);
+
+	for (i = 0; i < part_count; i++) {
+		depot->parts[i].head = NULL;
+		depot->parts[i].hash = 0;
+	}
+
+	kt_cache_init(&depot->stack_cache,
+		sizeof(kt_stack_depot_obj_t), stack_count);
```

Comment:
it is too wasteful to allocate 96 frames when the stack can be just 10 frames
just do a simple region allocator, bump the pointer to allocate kt_stack_depot_obj_t of necessary size